server/src/main/java/io/spine/server/migration/mirror/MirrorMigration.java

```diff
@@ -0,0 +1,186 @@
+     *         the storage provider
+     * @param aggClass
+     *         the class of an aggregate, {@code Mirror} projections of which are to be migrated
+     */
+    public MirrorMigration(ContextSpec contextSpec,
+                           StorageFactory factory,
+                           Class<A> aggClass) {
+
+        this.mirrorStorage = new MirrorStorage(contextSpec, factory);
+        this.entityRecordStorage = factory.createEntityRecordStorage(contextSpec, aggClass);
+        this.mapping = new MirrorMapping<>(aggClass);
+        this.aggregateType = AggregateClass.asAggregateClass(aggClass)
+                                          .stateTypeUrl()
+                                          .value();
+    }
+
+    /**
+     * Migrates {@linkplain Mirror} projections
+     * to the aggregate's {@linkplain EntityRecordStorage}.
+     */
+    public void run(MigrationSupervisor supervisor) {
+        supervisor.onMigrationStarted();
+
+        var step = proceed(supervisor);
+        while (step.getMigrated() != 0 && supervisor.shouldContinueAfter(step)) {
+            step = proceed(supervisor);
+        }
+
+        supervisor.onMigrationCompleted();
+    }
+
+    /**
+     * Fetches and processes the next batch.
+     */
+    private MigrationStep proceed(MigrationSupervisor supervisor) {
+        supervisor.onStepStarted();
+
+        var batchSize = supervisor.stepSize();
+        Collection<EntityRecordWithColumns<I>> entityRecords = new ArrayList<>(batchSize);
+        Collection<Mirror> migratedMirrors = new ArrayList<>(batchSize);
+
+        fetchNextBatch(batchSize)
+                .forEachRemaining(mirror -> {
+
+                    var entityRecord = mapping.toEntityRecord(mirror);
+                    var migratedMirror = mirror.toBuilder()
+                            .setWasMigrated(true)
+                            .build();
+
+                    entityRecords.add(entityRecord);
+                    migratedMirrors.add(migratedMirror);
+                });
+
+        entityRecordStorage.writeAll(entityRecords);
+        mirrorStorage.writeBatch(migratedMirrors);
+
+        var completedStep = MigrationStep.newBuilder()
+                .setMigrated(migratedMirrors.size())
+                .build();
+
+        supervisor.onStepCompleted(completedStep);
+        return completedStep;
+    }
+
+    private Iterator<Mirror> fetchNextBatch(int batchSize) {
+        var query = mirrorStorage.queryBuilder()
+                                 .where(Mirror.Column.aggregateType()).is(aggregateType)
+                                 .where(Mirror.Column.wasMigrated()).is(false)
+                                 .sortAscendingBy(Mirror.Column.wasMigrated())
+                                 .limit(batchSize)
+                                 .build();
+        var iterator = mirrorStorage.readAll(query);
+        return iterator;
+    }
+
+    @VisibleForTesting
+    EntityRecordStorage<I, S> entityRecordStorage() {
```

Comment:
As long as it's no `private`, please document it. 

Same goes for the method below.